MoreLinq/HasDuplicates.cs

```diff
@@ -0,0 +1,56 @@
+namespace MoreLinq
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    static partial class MoreEnumerable
+    {
+        /// <summary>
+        ///   Checks if sequence contains duplicates
+        /// </summary>
+        /// <param name="source">The sequence to check.</param>
+        /// <typeparam name="T">The type of the elements in the source sequence</typeparam>
+        /// <returns>true if any element of the sequence is duplicated, false otherwise</returns>
+        public static bool HasDuplicates<T>(this IEnumerable<T> source) => source?.HasDuplicates(EqualityComparer<T>.Default) ?? throw new ArgumentNullException(nameof(source));
```

Comment:
nit: everything after `=>` should be on the next line, for screen width concerns.